### PR TITLE
[Security] Deprecate simple_preauth and simple_form in favor of Guard

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php [hunk 1]
```diff
@@ -265,6 +267,10 @@ private function addFirewallsSection(ArrayNodeDefinition $rootNode, array $facto
                     ->canBeUnset()
                 ;
 
+                if ($factory instanceof SimplePreAuthenticationFactory || $factory instanceof SimpleFormFactory) {
+                    $factoryNode->setDeprecated(sprintf('The "%s" security listener is deprecated Symfony 4.2, use "guard" instead.', $name));
```
Comment (fabpot) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php]: "use Guard instead" to be consistent everywhere?

**Review by fabpot — APPROVED** [review]
small change for consistency

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/SimpleFormFactory.php [hunk 2]
```diff
@@ -18,14 +18,20 @@
...
 class SimpleFormFactory extends FormLoginFactory
 {
-    public function __construct()
+    public function __construct(bool $triggerDeprecation = true)
     {
         parent::__construct();
 
         $this->addOption('authenticator', null);
+
+        if (true === $triggerDeprecation) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/SimpleFormFactory.php]: ~`true === `~

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/SimplePreAuthenticationFactory.php [hunk 3]
```diff
@@ -18,9 +18,18 @@
...
 /**
  * @author Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * @deprecated since Symfony 4.2, use Guard instead.
  */
 class SimplePreAuthenticationFactory implements SecurityFactoryInterface
 {
+    public function __construct(bool $triggerDeprecation = true)
+    {
+        if (true === $triggerDeprecation) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/SimplePreAuthenticationFactory.php]: ~`true === `~

**Review by nicolas-grekas — APPROVED** [review]
(with minor comments)

**Comment by chalasr** [comment]
Comments addressed, thanks

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**